recipes/glad/2.x/conanfile.py

```diff
@@ -0,0 +1,158 @@
+
+    def validate(self):
+        if (self.options.gles1_version != "None" or self.options.gles2_version != "None") and self.options.egl_version == "None":
+            raise ConanInvalidConfiguration(f"{self.ref} Generating an OpenGLES spec requires a valid version of EGL")
+        if self.options.debug_layer and self.options.multicontext:
+            raise ConanInvalidConfiguration("The multicontext and debug layer options are incompatible")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if os.getenv("CONAN_CENTER_BUILD_SERVICE"):
+            # We need the Jinja2 package for Glad's code generation,
+            # but the Python found by CMake's FindPython may not have it installed.
+            # For CCI runners, we know that the Python running Conan will definitely have it.
+            tc.cache_variables["Python_EXECUTABLE"] = sys.executable
+
+        tc.cache_variables.update({
+                "GLAD_SOURCES_DIR": self.source_folder,
+                "GLAD_CONAN_LIB_TYPE": "SHARED" if self.options.shared else "STATIC",
+                "GLAD_CONAN_API": self._get_api(),
+        })
+        if self.options.extensions:
+            tc.cache_variables["GLAD_CONAN_EXTENSIONS"] = ";".join(str(self.options.extensions).split(","))
+
+        if not self.options.no_loader:
+            tc.cache_variables["GLAD_CONAN_LOADER"] = "LOADER"
+
+        if self.options.debug_layer:
+            # wraps all gl calls in a debugging layer - can be used independently of the
+            # conan build type
+            # See https://github.com/Dav1dde/glad/wiki/C#debugging
+            tc.cache_variables["GLAD_CONAN_DEBUG"] = "DEBUG"
+
+        if self.options.multicontext:
+            tc.cache_variables["GLAD_CONAN_MX"] = "MX"
+
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=Path(self.source_folder).parent)
+        cmake.build()
+
+    def _get_api(self):
+        def api(spec):
+            for api_name, api_version in spec.items():
+                if api_version == "None":
+                    continue
+                if api_name == "gl":
+                    yield f"{api_name}:{self.options.gl_profile}={api_version}"
+                else:
+                    yield f"{api_name}={api_version}"
+
+        spec_api = {
+            "gl": self.options.gl_version,
+            "gles1": self.options.gles1_version,
+            "gles2": self.options.gles2_version,
+            "glsc2": self.options.glsc2_version,
+            "egl": self.options.egl_version,
+            "glx": self.options.glx_version,
+        }
+
+        wgl_version = self.options.get_safe("wgl_version")
+        if wgl_version:
+            spec_api["wgl"] = wgl_version
+
+        return ";".join(api(spec_api))
```

Comment:
I think that it's easier to read:

```suggestion
def _get_api(self):
    spec_api = {
        "gl": self.options.gl_version,
        "gles1": self.options.gles1_version,
        "gles2": self.options.gles2_version,
        "glsc2": self.options.glsc2_version,
        "egl": self.options.egl_version,
        "glx": self.options.glx_version,
        "wgl": self.options.get_safe("wgl_version", "None")
    }
    return ";".join(
        f"{name}:{self.options.gl_profile}={version}" if name == "gl" else f"{name}={version}"
        for name, version in spec_api.items()
        if version != "None"
    )
```